### PR TITLE
Add geocoding/reverse geocoding to create/update UI

### PR DESCRIPTION
**Pull request by benvogler**
This update automatically geocodes addresses/reverse geocodes coordinates when they're input on the create/update form, then displays the coordinates/address/city/state to the user (city/state un-editable) and sends them in the update to the server so they're stored.

No need to geocode/reverse geocode on the backend for singular create/update now, only for bulk

Address

Coordinates

### CODE REVIEW

File: client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js [hunk 1]
```diff
@@ -503,9 +509,13 @@ export default class CreateOrUpdateForm extends React.Component {
             photoSphereImages: photoSphereImages.map(photoSphereImage => photoSphereImage.url),
             newIsTemporary: isTemporary.value,
             dateSelectValue: dateSelectValue,
-            imagesForUpdate: imagesForUpdate
+            imagesForUpdate: imagesForUpdate,
+            newCity: city,
+            newState: state
         };
 
+        console.log(updateForm);
```
Comment (njd9419) [inline client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js:517]: Minor: Debug print

File: client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js [hunk 2]
```diff
@@ -606,6 +641,22 @@ export default class CreateOrUpdateForm extends React.Component {
...
+            if (component.types.includes('locality')) {
+                city = component.long_name;
+            }
+            if (component.types.includes('administrative_area_level_1')) {
+                state = component.short_name;
+            }
+        }
+        address = result.formatted_address;
+        this.setState({city, state, address: {...this.state.address, value: address}});
+        console.log({city, state, address: {...this.state.address, value: address}});
```
Comment (njd9419) [inline client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js:659]: Minor: Debug print

File: client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js [hunk 3]
```diff
@@ -934,7 +985,8 @@ export default class CreateOrUpdateForm extends React.Component {
     render() {
         const { showingAdvancedInformation, dateSelectValue, datePickerCurrentDate, title, address, latitude,
             longitude, year, month, artist, description, inscription, references, imageUploaderKey, materials,
-            imagesForUpdate, isTemporary, locationType, photoSphereImagesForUpdate, photoSphereImages } = this.state;
+            imagesForUpdate, isTemporary, locationType, photoSphereImagesForUpdate, photoSphereImages,
+            city, state} = this.state;
```
Comment (njd9419) [inline client/src/components/CreateOrUpdateForm/CreateOrUpdateForm.js:989]: Minor: There should be a space after `state`

File: client/src/components/Monument/Update/MonumentUpdate.js [hunk 4]
```diff
@@ -292,6 +292,16 @@ export default class MonumentUpdate extends React.Component {
...
 
+            /* City */
+            (this.didAttributeChange(oldMonument.city, update.newCity)) ?
+                changedAttributes.push(<AttributeChange attributeLabel="City" oldAttribute={oldMonument.city} newAttribute={update.newCity} key="city"/>) :
+                unchangedAttributes.push(<AttributeChange attributeLabel="City" oldAttribute={oldMonument.city} newAttribute={update.newCity} didChange={false} key="city"/>);
+
+            /* State */
+            (this.didAttributeChange(oldMonument.state, update.newState)) ?
+                changedAttributes.push(<AttributeChange attributeLabel="City" oldAttribute={oldMonument.state} newAttribute={update.newState} key="state"/>) :
+                unchangedAttributes.push(<AttributeChange attributeLabel="City" oldAttribute={oldMonument.state} newAttribute={update.newState} didChange={false} key="state"/>);
```
Comment (njd9419) [inline client/src/components/Monument/Update/MonumentUpdate.js:303]: Both of these should have `attributeLabel` = `"State"`

Comment (benvogler) [reply client/src/components/Monument/Update/MonumentUpdate.js:303]: Good catch

**Pull request merged**